engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/integer/ModNode.java

```diff
@@ -7,21 +7,42 @@
 import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
 import org.enso.interpreter.node.expression.builtin.number.utils.ToEnsoNumberNode;
 import org.enso.interpreter.runtime.EnsoContext;
-import org.enso.interpreter.runtime.builtin.Builtins;
 import org.enso.interpreter.runtime.error.DataflowError;
-import org.enso.interpreter.runtime.error.PanicException;
 import org.enso.interpreter.runtime.number.EnsoBigInteger;
 
-@BuiltinMethod(type = "Big_Integer", name = "%", description = "Big integer modulo division.")
+@BuiltinMethod(type = "Integer", name = "%", description = "Modulo division of numbers.")
 public abstract class ModNode extends Node {
   private @Child ToEnsoNumberNode toEnsoNumberNode = ToEnsoNumberNode.create();
 
-  abstract Object execute(EnsoBigInteger self, Object that);
+  abstract Object execute(Object self, Object that);
 
   static ModNode build() {
     return ModNodeGen.create();
   }
 
+  @Specialization
+  Object doLong(long self, long that) {
+    try {
+      return self % that;
+    } catch (ArithmeticException e) {
+      return DataflowError.withoutTrace(
+          EnsoContext.get(this).getBuiltins().error().getDivideByZeroError(), this);
+    }
+  }
+
+  @Specialization
+  double doDouble(long self, double that) {
+    // No need to trap, as floating-point modulo returns NaN for division by zero instead of
+    // throwing.
+    return self % that;
+  }
+
+  @Specialization
+  long doBigInteger(long self, EnsoBigInteger that) {
+    // No need to trap, as 0 is never represented as an EnsoBigInteger.
+    return self;
+  }
```

Comment:
I'd like to check if that is still true.

This definitely used to be true, but now we not only have BigIntegers that come from values too large to represent. We also have BigIntegers that simply come from Java or Python via interop.

What will happen if I construct a Java BigInteger and pass it to Enso and do `23 / my_java_bigint`? Please add such a test to the test suite to verify this is correct.

Reply:
Good observation. Fixed in bf475e1aaa99720c0c75a58b0af7315a38abeb15.